src/sql/workbench/parts/notebook/browser/models/notebookModel.ts

```diff
@@ -670,6 +662,10 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (kernel.info) {
 			this.updateLanguageInfo(kernel.info.language_info);
 		}
+		this._kernelChangedEmitter.fire({
+			newValue: kernel,
+			oldValue: undefined
```

Comment:
You don't want to get the existing kernel value and pass that as the oldValue?

Reply:
Actually, no. It's part of the APIs but we don't use it.

Reply:
I'd suggest leaving a comment explaining that then, since if I came across this I would think it's a bug at first glance.